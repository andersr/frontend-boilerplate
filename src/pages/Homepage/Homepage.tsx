import React from 'react'
import { Link } from 'react-router-dom'

import { SamplePngFile } from '../../assets/images'
import { CurrentCount } from '../../components'
export const HOMEPAGE_TEST_ID = 'homepage'

export function Homepage() {
  return (
    <div data-testid={HOMEPAGE_TEST_ID}>
      <h1>Homepage</h1>
      <div>
        <Link to="/about">About Us</Link>
      </div>
      <CurrentCount />
      <img src={SamplePngFile} alt="Sample Image" />
    </div>
  )
}
