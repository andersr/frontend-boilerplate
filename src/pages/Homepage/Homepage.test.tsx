import React from 'react'
import { MemoryRouter } from 'react-router'
import renderer from 'react-test-renderer'

import { Homepage } from './Homepage'

describe('Homepage', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Homepage />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
