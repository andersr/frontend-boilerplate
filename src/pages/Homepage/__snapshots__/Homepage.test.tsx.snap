// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders correctly 1`] = `
.c0 {
  height: 20px;
  background-color: green;
}

<div
  data-testid="homepage"
>
  <h1>
    Homepage
  </h1>
  <div>
    <a
      href="/about"
      onClick={[Function]}
    >
      About Us
    </a>
  </div>
  <div
    className="c0"
    data-testid="currentCount"
  >
    Count: 
    0
     
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <img
    alt="Sample Image"
    src="test-file-stub"
  />
</div>
`;
